engines/scumm/sound.cpp

```diff
@@ -574,6 +574,17 @@ void Sound::playSound(int soundID) {
 			stream = plainStream;
 		}
 
+		// The real meaning of this byte is currently unknown, but we assume that
+		// it's meant to interrupt any identical sound which isn't fully done
+		// playing yet when it's unset (Player_V3A::startSound() does something
+		// similar for Indy3 Amiga). A good test for this behavior is to ring the
+		// boxing bell at the start of the game; in the original interpreter it
+		// rings 3 times in a row, but if we don't do this the second bell sound
+		// is never heard.
+		// TODO: check this against actual Indy3 Macintosh disasm (anyone?).
```

Comment:
If we are not sure about this change, and we need a full playthrough to test it, it would be better to restrict this workaround only for the boxing room, unless another case is found

Reply:
Personally I would be for a full playthrough, in order to prove that this is the correct behavior at a driver level.

I think of these changes as guess work which is very educated, instead of naive, and it would be a real shame to restrict what could very potentially be the correct behavior to "only" a workaround. Beside, we should try avoiding the latters as much as we can 😄

Reply:
> If we are not sure about this change, and we need a full playthrough to test it, it would be better to restrict this workaround only for the boxing room, unless another case is found

As you wish! I *am* willing to do a full playthrough of Indy3 Macintosh to check for any regression, and I'm willing to start it right now! (I'm also quite confident in that change since Indy3 Amiga has a similar behavior confirmed by disasm AFAICS, and since I'm only stopping any current sound with the *same* soundID).

But if we want to be extra careful, we can just limit this change to this room yeah.

Reply:
I'm also very confident in the change for the same reason (it's not unusual to see similarly structured data formats for different target platforms), that's why IMHO this should *not* restricted as a workaround.

Reply:
Thanks. I’ve done that full gameplay again, and I’m not hearing any other difference, except for the thunder sound when you’re outside the windows of Castle Brunwald.

The effect is now different and I’m not sure if the new or the old effect is the right one. They both sound a bit weird.

Putting this on hold until I can check this part against the behavior of the original Macintosh interpreter.